### PR TITLE
Fix sak

### PR DESCRIPTION
**Pull request by esiragusa**
The tool was (ridiculously) implemented in a wrong way and the test was wrong too...

PS I didn't bother to fix all the issues, e.g. there's still a segfault when reading input from fasta and writing output to stdout (in fastq).

### CODE REVIEW

**Comment by h-2** [comment]
Ok, I have no clue about the tool and can't really judge your changes from scanning the diff, but if @rrahn doesn't either, I will just merge.

File: apps/sak/sak.cpp [hunk 1]
```diff
@@ -386,78 +392,39 @@ int main(int argc, char ** argv)
...
-                reverse(quals);
-                infixEnd = length(seq) - infixEnd;
-                infixBegin = length(seq) - infixBegin;
-                std::swap(infixEnd, infixBegin);
-
-                writeRecord(outFile, id, infix(seqCopy, infixBegin, infixEnd), infix(quals, infixBegin, infixEnd));
-            }
-            else
-                writeRecord(outFile, id, infix(seq, infixBegin, infixEnd), infix(quals, infixBegin, infixEnd));
+            seqan::Dna5String seqCopy = seq;
```
Comment (rrahn) [inline apps/sak/sak.cpp]: Is the copy even necessary?

File: apps/sak/sak.cpp [hunk 2]
```diff
@@ -386,78 +392,39 @@ int main(int argc, char ** argv)
...
+            infixBegin = length(seq) - infixBegin;
+            std::swap(infixEnd, infixBegin);
+
+            writeRecord(outFile, id, infix(seqCopy, infixBegin, infixEnd), infix(quals, infixBegin, infixEnd));
         }
-
-        // Advance counter idx.
-        idx += 1;
+        else
+            writeRecord(outFile, id, infix(seq, infixBegin, infixEnd), infix(quals, infixBegin, infixEnd));
```
Comment (rrahn) [inline apps/sak/sak.cpp]: Please, add braces around else, if the associated if also has braces.😜

**Comment by rrahn** [comment]
Ok, except the minor remarks, it looks good to me.

**Comment by esiragusa** [comment]
@rrahn your remarks refer to code that I didn't change...

File: apps/sak/sak.cpp [hunk 3]
```diff
@@ -369,12 +369,18 @@ int main(int argc, char ** argv)
     // -----------------------------------------------------------------------
     startTime = seqan::sysTime();
 
-    unsigned idx = 0;
+
```
Comment (rrahn) [inline apps/sak/sak.cpp]: empty line

**Comment by rrahn** [comment]
I see. But I found an empty line. 😉

**Comment by rrahn** [comment]
@esiragusa I can merge it, but you have to sign the CLA first 😉

**Comment by rrahn** [comment]
@esiragusa Before I forgot, can you please update the changelog if there was resolved a major issue?

**Comment by h-2** [comment]
we are only merging real "hotfixes" master now, i.e. regressions or breakages.

**Comment by esiragusa** [comment]
WTF??

**Comment by h-2** [comment]
@esiragusa No reason to become angry. We have changed the policy of our releases:
* a minor release every 3 months
* no breakage of documented API allowed for minor releases
* minor releases prepared on develop branch, but merged to master before release
* patchlevel releases from master are only performed in the case that there was a mistake in the release procedures or a serious issue, like a regression or an API breakage
→ all regular PRs only go to develop

Since you still have PR to develop this will be included in the next release provided that @rrahn merges it (I don't know what this depends on).

**Pull request closed without merging**